### ISSUE TITLE
Remove widgets?

### ISSUE THREAD

**Issue by DannyOutlaw** [pojome/elementor#597]
Is there a function to remove unwanted widgets?  Say I have no need for the google maps widget and want to remove it from elementor.

Can I do that?

**Comment by arielk** [comment]
Currently it is not possible, we're considering it for the developers API

**Comment by spacemano** [comment]
Hello,

Try this code

```
add_action('elementor/widgets/widgets_registered', 'custom_unregister_elementor_widgets');

function custom_unregister_elementor_widgets($obj){
	$obj->unregister_widget_type('image'); // This hides the Image widget for example
}
```

You can see the list of all widgets in elementor/includes/managers/widgets.php in the $build_widgets_filename array.

**Comment by spacemano** [comment]
It doesn't seem to be possible without editing some of their code, which would be lost on plugin update.

It should have worked by updating an option value to something like this
`update_option( 'elementor_pro_active_modules', array( 'countdown' => false, 'flip-box' => false ) );`
Where 'countdown' and 'flip-box' are the names of the widgets you want to disable, for example.
BUT this is ignored for widgets that are set to "_required_", and all of them are...

This can be found in elementor-pro/includes/modules-manager.php

If you're not going to update the plugin you can just comment out the lines with the widgets you aren't planning to use ( $modules array in __construct() ). But it seems like disabling some of them may lead to javascript errors, so be careful. It seems like some of them might really be required (or relied upon in the javascript files)

**Comment by macolo** [comment]
@arielk Hi! I wonder if this is still not working for elementor pro widgets?

**Comment by bainternet** [comment]
@macolo 

you can unregister widgets if you are using the API correctly [see here](https://github.com/pojome/elementor/issues/3136#issuecomment-361602293)

as for @spacemano suggestion, as far as I know, it shouldn't work.

**Comment by bdkoder-official** [comment]
### Remove Unwanted Widget/Widgets

```php

$elementor_widget_blacklist = [
    'your-widget',
    'your-widget-name-2'
];

// Remove all the Elementor Free & Pro Widgets listed above
add_action('elementor/widgets/register', function ($widgets_manager) {
    global $elementor_widget_blacklist;
    foreach ($elementor_widget_blacklist as $widget_name) {
        $widgets_manager->unregister_widget_type($widget_name);
    }
}, 9999);
```

**Comment by bdkoder** [comment]
### Remove Unwanted Widget/Widgets

```php

$elementor_widget_blacklist = [
    'your-widget',
    'your-widget-name-2'
];

// Remove all the Elementor Free & Pro Widgets listed above
add_action('elementor/widgets/register', function ($widgets_manager) {
    global $elementor_widget_blacklist;
    foreach ($elementor_widget_blacklist as $widget_name) {
        $widgets_manager->unregister_widget_type($widget_name);
    }
}, 9999);
```

**Comment by nicholaszein** [comment]
Hello there!

We have great news! 🙌 

### 📢 We're happy to announce that this feature was released with Elementor v3.18.0! 🥳 

You can now use the Elements Manager to enable/disable widgets.

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
